rfcs/0001-api-v2/resources/Semaphore API Design Guide.md

```diff
@@ -0,0 +1,362 @@
+## 
+
+[Introduction](#introduction)
+
+[Resources](#resources)
+
+[Resource-Oriented Design](#resource-oriented-design)
+
+[Nested Resources](#nested-resources)
+
+[Endpoint Naming](#endpoint-naming)
+
+[Response](#response)
+
+[Response Format](#response-format)
+
+[Object Instead of IDs](#object-instead-of-ids)
+
+[Standard Methods](#standard-methods)
+
+[Create a Resource](#create-a-resource)
+
+[Read a Resource](#read-a-resource)
+
+[Update a Resource](#update-a-resource)
+
+[Delete a Resource](#delete-a-resource)
+
+[Custom Methods](#custom-methods)
+
+[Behavior of Custom Methods](#behavior-of-custom-methods)
+
+[Request and Response Structure](#request-and-response-structure)
+
+[Examples of Custom Methods](#examples-of-custom-methods)
+
+[Encouraging the Use of Standard Methods](#encouraging-the-use-of-standard-methods)
+
+[Error Handling](#error-handling)
+
+[Error Response Format](#error-response-format)
+
+[Writing Effective Error Messages](#writing-effective-error-messages)
+
+[Error Codes](#error-codes)
+
+[HTTP Status Codes](#http-status-codes)
+
+[Best Practices for Error Handling](#best-practices-for-error-handling)
+
+[Standard Fields](#standard-fields)
+
+[Naming Convention: snake\_case](#naming-convention:-snake_case)
+
+[Timestamps](#timestamps)
+
+## Introduction {#introduction}
```

Comment:
Do we need the {#...} part in the headers?

Reply:
I'll just use this comment thread to make a note about the formatting:
- Table of contents should be re-formatted.
- Code blocks in this file should be properly formatted too.